File: leetcode/2602/260222-868-binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        while (n % 2 == 0) {
            n /= 2;
        }

        int ret = 0;
        while (n % 2 == 1) {
            n >>= 1;
            if (n == 0) return ret;
            int t = 1;
            while (n % 2 != 1) {
                n >>= 1; t++;
            }
            ret = max(ret, t);
        }
        return ret;
    }
};

File: leetcode/2602/260222-868-binary-gap.rs
impl Solution {
    pub fn binary_gap(mut n: i32) -> i32 {
        let mut ret = 0;

        while n % 2 != 1 {
            n >>= 1;
        }

        loop {
            n >>= 1;
            if n == 0 { break ret; }

            let mut t = 1;
            while n % 2 != 1 {
                n >>= 1; t += 1;
            }

            ret = if t > ret { t } else { ret };
        }
    }
}
